Adding demo stress test + changing CI pipelines for Deepseek V3

yalrawwashTT opened the pull request:
### Ticket
#30035

### Problem description
Add test for deepseek demo and execute in CI

### What's changed
Test has been added that compares 32 user output for demo compared against a json file with the pre-generated outputs.

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/19106594408) CI passes
- [ ] [Deepseek demo](https://github.com/tenstorrent/tt-metal/actions/runs/19105414489) CI passes

yalrawwashTT committed:
adding demo test

yalrawwashTT committed:
updating paths for CI

pprajapatiTT commented on models/demos/deepseek_v3/demo/test_demo_exact_word_match.py:
```diff
@@ -0,0 +1,273 @@
+_GEN_HDR_RE = re.compile(r"^Generation\[(\d+)\]:")
+
+
+def extract_all_generations(stdout: str):
```
Printing mechanism may change, we don't want to rely on it.

call run_demo() and parse return list of results

yalrawwashTT committed:
calling run_demo as opposed to subprocess and passing max_tokens as a parametrization

pprajapatiTT commented on models/demos/deepseek_v3/demo/test_demo_exact_word_match.py:
```diff
@@ -0,0 +1,193 @@
+    Defaults to case/punctuation-sensitive exact match (WER==0); see flags above.
+    """
+    # cache_dir = tmp_path / "cache"
+    cache_dir = Path("/proj_sw/user_dev/deepseek-v3-cache")
```
use env variable, like used in other tests and demo

pprajapatiTT commented on models/demos/deepseek_v3/demo/test_demo_exact_word_match.py:
```diff
@@ -0,0 +1,193 @@
+        pytest.fail(message)
+
+    # If we get here, all prompts passed
+    assert True
```
is this really needed?

yalrawwashTT committed:
updating CI paths and finalizing

yalrawwashTT committed:
moving files into tests directory so they are run by CI

yalrawwashTT committed:
removing setup_metal.sh

yalrawwashTT committed:
adding import for CI paths

yalrawwashTT committed:
running in diff CI pipeline

yalrawwashTT committed:
adding fast fail & stress test functionality

yalrawwashTT committed:
updating CI paths

yalrawwashTT committed:
renaming num_runs to repeat_batches to maintain consistency with other parts of code base

yalrawwashTT committed:
updating timeout so that CI test can go to completeion

pprajapatiTT commented on tests/scripts/tg/run_tg_deepseek_tests.sh:
```diff
@@ -0,0 +1,84 @@
+  export FAKE_DEVICE=TG
+  pytest -n auto \
+    models/demos/deepseek_v3/demo/test_demo_exact_word_match.py \
+    --timeout 1000
```
instead of 1000 sec, i think, we need 25 mins timeout for 1 repeat batch. Since we plan to have 2 repeat batches, (and only once loading time) we should be good with 35-40 mins. But try some values around 35-40 mins and see how that goes. We dont want to use bigger timeouts too in CI.

pprajapatiTT commented on tests/scripts/run_tests.sh:
```diff
@@ -196,8 +196,14 @@ demos_tg_device() {
-    ./tests/scripts/tg/run_tg_demo_tests.sh --model "$model"
+    case "$model" in
+        deepseek_v3)
+            ./tests/scripts/tg/run_tg_deepseek_tests.sh --model "$model"
```
Well, we are already checking the model name in case, so we don't need --model arg again.

yalrawwashTT committed:
Merge branch 'main' into deepseek-demo

yalrawwashTT committed:
Merge branch 'main' into deepseek-demo

yalrawwashTT committed:
Merge branch 'main' into deepseek-demo

yalrawwashTT committed:
Remove FAKE_DEVICE export from test script

Removed the export of FAKE_DEVICE for the test run.

yalrawwashTT committed:
Increase pytest timeout from 1000 to 2700 seconds

yalrawwashTT committed:
Remove redundant model argument in deepseek tests

yalrawwashTT committed:
Decrease timeout for Galaxy DeepSeek v3 demo tests

Reduced timeout for Galaxy DeepSeek v3 demo tests from 60 to 45 minutes.

yalrawwashTT committed:
Change repeat_batches parameter from 1 to 2

yalrawwashTT committed:
Only run deepseek test to speed up CI run.

Updated timeout for Galaxy DeepSeek v3 demo tests to 60 seconds and commented out other test configurations.

Commit:
updating output json to ensure passing test + verified passing demo test locally

yalrawwashTT committed:
Increase pytest timeout from 2700 to 5100 seconds in bash script

yalrawwashTT committed:
Update timeout for Galaxy DeepSeek v3 demo tests to 85 min (based on local test duration)

Increased timeout for Galaxy DeepSeek v3 demo tests from 60 to 85 seconds.

yalrawwashTT committed:
removing word-to-word check for now

yalrawwashTT committed:
changing word to word check to top1/5 token accuracy check with teacher forcing
